src/slotted_page.rs

```diff
@@ -203,7 +229,24 @@ impl SlottedPageMmap {
             transmute_from_u8(&mmap[0..size_of::<SlottedPageHeader>()]);
         let header = header.clone();
         let path = path.to_path_buf();
-        Some(SlottedPageMmap { path, header, mmap })
+        let mut page = SlottedPageMmap {
+            path,
+            header,
+            mmap,
+            deleted_slots: PriorityQueue::new(),
+        };
+
+        // Populate the deleted slots queue
+        let mut deleted_slots = PriorityQueue::new();
+        for (slot, slot_id) in page.iter_slots_starting_from(0).zip(0u32..) {
```

Comment:
this could be expensive on large page with a lot of slots, might make sense to pre-compute it